lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -254,89 +170,51 @@ export class UserAgentApplication {
-      } = {}) {
-      const {
-          validateAuthority = true,
-          cacheLocation = "sessionStorage",
-          redirectUri = () => window.location.href.split("?")[0].split("#")[0],
-          postLogoutRedirectUri = () => window.location.href.split("?")[0].split("#")[0],
-          logger = new Logger(null),
-          loadFrameTimeout = 6000,
-          navigateToLoginRequestUrl = true,
-          state = "",
-          isAngular = false,
-          unprotectedResources = new Array<string>(),
-          protectedResourceMap = new Map<string, Array<string>>(),
-          storeAuthStateInCookie = false
-      } = options;
-
-    // TODO: The configuration object will take care of many of these assignments
-    this.loadFrameTimeout = loadFrameTimeout;
-    this.clientId = clientId;
-    this.validateAuthority = validateAuthority;
-
-    // TODO: Replace string with constant
-    this.authority = authority || "https://login.microsoftonline.com/common";
-    this._navigateToLoginRequestUrl = navigateToLoginRequestUrl;
-    this._state = state;
-    this._isAngular = isAngular;
-    this._unprotectedResources = unprotectedResources;
-    this._protectedResourceMap = protectedResourceMap;
-    this._redirectUri = redirectUri;
-    this._postLogoutredirectUri = postLogoutRedirectUri;
-    this._logger = logger;
-    this.storeAuthStateInCookie = storeAuthStateInCookie;
-
-    // TODO: This should be replaced with two different callbacks for success and error cases
-    this._tokenReceivedCallback = tokenReceivedCallback;
-
-    // Track login and acquireToken in progress
-    this._loginInProgress = false;
-    this._acquireTokenInProgress = false;
-
-    // TODO: This should be replaced with cache object, typescript checking for "localStorage" and "sessionStorage" values
-    this._cacheLocation = cacheLocation;
-    if (!this._cacheLocations[cacheLocation]) {
-      throw new Error("Cache Location is not valid. Provided value:" + this._cacheLocation + ".Possible values are: " + this._cacheLocations.localStorage + ", " + this._cacheLocations.sessionStorage);
-    }
-    this._cacheStorage = new Storage(this._cacheLocation); //cache keys msal
+  constructor(configuration: Configuration, callback: tokenReceivedCallback) {
+
+    // Set the Configuration
+    this.config = configuration;
+
+    // Set the callback
+    this.tokenReceivedCallback = callback;
+
+
+    this.logger = this.config.system.logger;
+    this.clientId = this.config.auth.clientId;
+    this.inCookie = this.config.cache.storeAuthStateInCookie;
+
+    // if no authority is passed, set the default: "https://login.microsoftonline.com/common"
+    this.authority = this.config.auth.authority || DEFAULT_AUTHORITY;
+
+    // track login and acquireToken in progress
+    this.userLoginInProgress = false;
+    this.acquireTokenInProgress = false;
+
+    // cache keys msal - typescript throws an error if any value other than "localStorage" or "sessionStorage" is passed
```

Comment:
question, do we throw anything if typescript does not? not all consumers are using typescript

Reply:
In that case may be we should not make this language specific?

Reply:
When you say language specific what do you mean? Ts vs Js?
I think we should write code for the lowest common denominator we compile down to (es5) and treat the typesafety of typescript as sugar on top for those using the msal lib in vscode 

use case: 

I am using the msal.js file in my spa, downloaded from cdn.  I put in the wrong storage type,  maybe storage: 'database'
Msal.js throws a ClientConfigurationError

usecase2
I am using the msal lib in my project, and in vscode when I try and set storage to 'database' It throws an angry little redline and I fix this before committing code.  (keep in mind here that even then, not all people who use are lib are using typescript to check types before deploying, so it is a dev convenience and we still require the ClientConfigurationError)

Reply:
Yes. That is what I meant. I initially coded that way but got carried away with the convenience of TS doing it for me. Let me make these changes and send out for review sometime today to close this PR and merge it to the API branch.